### DISCUSSION TITLE
Update a git submodule in the GitHub UI?

### DISCUSSION THREAD

**Discussion by probonopd** [community/community#15079] [Other Feature Feedback, Questions, & Ideas]
I frequently edit files in the GitHub UI, but I didn't find a way to update a git submodule in the same way. Which sends me back to the command line.

Bummer!

I wish there was an _easy_ way to update a git submodule in the GitHub UI, by pressing the "Edit" button and entering `c58ac7`:

![image](https://user-images.githubusercontent.com/2480569/163712244-10fc4a31-699d-431d-a7e0-8e3060d5893e.png)

Or is there?

**Comment by davidstaheli** [comment]
Thanks for posting about this, @probonopd. We on GitHub's repositories team agree this would be very nice to have. I've added it to our backlog to be prioritized.

In the meantime, one option that might help (if you're using github.com) is to type '.' (period) when you're viewing a repository in the GitHub UI. That will launch a github.dev editor based on VS Code. That editor has submodule support that might make it easier to do submodule file edits. See these VS Code feature announcements:

- [Nested Git repo support](https://code.visualstudio.com/updates/v1_72#_source-control) - detects and displays nested Git submodules.
- [Git submodule support](https://code.visualstudio.com/updates/v1_20#_git-submodules)
- [Git: Recursive clone command](https://code.visualstudio.com/updates/v1_51#_git-recursive-clone-command) - includes submodules

**Reply by dragoncoder047** [reply]
> That editor has submodule support that might make it easier to do submodule file edits. See these VS Code feature announcements:

I found this discussion because I had the same problem... I want to do some quick edits to a project with submodules using the github.dev editor (the `.` shortcut), but when I open to the submodule, the submodule's folder is correctly shown in blue with an 'S' to indicate it's a submodule, but a) the folder is empty, and b) the submodule doesn't show up in the source control pane.

None of the provided links gave me any help to either update the submodule or even show its files.

@davidstaheli what were you intending to refer to in those links that might provide help in this regard?

**Reply by TWiStErRob** [reply]
I tried `.` and confirm @dragoncoder047's findings, the web editor has no submodule support other than that `S`. We need to open a Terminal in VS Code web, which requires Codespaces. Even there, we have to do the full console process:
```bash
git submodule init
git submodule update
cd my/sub/module
git checkout whatever-branch
cd ../../..
git add my/sub/module
git commit -m "Updated my/sub/module"
git push
```

Don't forget to terminate the Codespaces instance with Ctrl/Cmd+Shift+P by entering and selecting "Stop Current Codespaces instance".

**Reply by IslamIssa1997** [reply]
thank you i have manage to update a submodule in my phone kernel tree without clone using your comment

**Comment by shaohef** [comment]
Can I update the submodule in the GitHub UI directly? 
I just want to update submodule, IMHO, ne need to pull the whole repo to local, update and then push.

**Comment by oscarbg** [comment]
+1

**Comment by tiliarou** [comment]
@davidstaheli would you say this feature is gonna come in the coming months or are we talking about years instead ?

**Reply by TWiStErRob** [reply]
> added it to our backlog to be prioritized

Means it's in the backlog, and it may eventually get to the top, or never. [ref](https://medium.com/product-love/the-scrum-backlog-is-where-features-go-to-die-b1336ff707aa).

**Reply by tiliarou** [reply]
> > added it to our backlog to be prioritized
> 
> Means it's in the backlog, and it may eventually get to the top, or never. [ref](https://medium.com/product-love/the-scrum-backlog-is-where-features-go-to-die-b1336ff707aa).

Alright, thanks for the reply.
That's very unfortunate (that's the polite version) that the web UI isn't seen as an important topic by Github or MS... It would support growth further.
I hope the article you linked don't say it all... (then if it is, what's even the point of having a backlog if we let it die and eventually flush it without addressing topics? 😅)

**Reply by TWiStErRob** [reply]
Btw, they're [definitely not resting, there's a lot going on](https://github.com/orgs/github/projects/4247), but as I scan through it, it seems mostly to increase the amount of money they can get from customers.
